### PR TITLE
ci: Manual Release workflow [ci skip]

### LINKED ISSUE #2097: Create a Github Action which allows use to patch Micronaut Launch web application without releasing artefacts or CLI
we could create a manual trigger.

CleanShot 2023-10-02 at 13 14 09@2x

### PR DESCRIPTION
**Pull request by sdelamo**
Close #2097

Uses a docker tag `latest` to release `analytics` and `starter` to Google Cloud Run.

### CODE REVIEW

File: .github/workflows/release-manually.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+      - uses: gradle/wrapper-validation-action@v1
+      - name: Set up JDK
+        uses: actions/setup-java@v3.11.0
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - name: projectVersion is not a snapshot
+        run: ./gradlew projectVersionNotSnapshot
```
Comment (timyates) [inline .github/workflows/release-manually.yml:19]: so, prior to running this, you need to manually set `micronautCoreVersion` (core), `micronautVersion` (platform) and `projectVersion` manually to non SNAPSHOT versions in a commit?

File: .github/workflows/release-manually.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - name: Set the current release version
+        id: release_version
+        run: |
+          release_version=${GITHUB_REF:11}
+          sed -i "s/^projectVersion.*$/projectVersion\=${release_version}/" gradle.properties
```
Comment (timyates) [inline .github/workflows/release-manually.yml]: And then this changes the project version (after we have checked it isn't SNAPSHOT above)

File: .github/workflows/release-manually.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+          token: ${{ secrets.GH_TOKEN }}
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - name: Set the current release version
+        id: release_version
+        run: |
+          release_version=${GITHUB_REF:11}
```
Comment (timyates) [inline .github/workflows/release-manually.yml]: Assuming we're running this off a branch, won't GITHUB_REF be

```
refs/heads/4.1.x
```

Which means `release_version` will be `4.1.x`

Is that what we want?

**Review by timyates — CHANGES_REQUESTED** [review]

File: .github/workflows/release-manually.yml (hunk 1 above)
Comment (sdelamo) [reply .github/workflows/release-manually.yml:19]: yes. Do a commit run the workflow, revert the commit.

File: .github/workflows/release-manually.yml (hunk 3 above)
Comment (sdelamo) [reply .github/workflows/release-manually.yml]: removed https://github.com/micronaut-projects/micronaut-starter/pull/2104/commits/511e786db0d5039d4ff7dc175dcd019af0aa1e18

File: .github/workflows/release-manually.yml (hunk 2 above)
Comment (sdelamo) [reply .github/workflows/release-manually.yml]: removed https://github.com/micronaut-projects/micronaut-starter/pull/2104/commits/511e786db0d5039d4ff7dc175dcd019af0aa1e18

File: .github/workflows/release-manually.yml (hunk 1 above)
Comment (timyates) [reply .github/workflows/release-manually.yml:19]: Should we add the instructions on how to use this to the top of this yaml as a comment?

Comment (sdelamo) [reply .github/workflows/release-manually.yml:19]: see https://github.com/micronaut-projects/micronaut-starter/pull/2104/commits/76b4c212c1c039bac516035ceb8b3c4663232ebc

**Review by timyates — APPROVED** [review]

**Pull request merged**